docs/project_setup_and_build.md

```diff
@@ -0,0 +1,229 @@
+inside the `project_dir/build/junit-reports` directory.
+1. Finally, `bigflow build-package` runs the `bdist_wheel` setup tools command. It generates a `.whl` package that you can
+upload to `pypi` or install locally - `pip install your_generated_package.whl`.
+
+Go to the `docs` project and run the `bigflow build-package` command to observe the result. Now you can install the 
+generated package using `pip install dist/examples-0.1.0-py3-none-any.whl`. After you install the `.whl` file, you can
+run jobs and workflows from the `docs/examples` package. They are now installed in your virtual env, so you can run them
+anywhere in your directory tree, for example:
+
+```shell script
+cd /tmp
+bigflow run --workflow hello_world_workflow --project-package examples
+```
+
+You probably won't use this command very often, but it's useful for debugging. Sometimes you want to see if your project
+works as you expect in the form of a package (and not just as a package in your project directory).
+
+### Project versioning
+
+Artifacts, like Python packages and Docker images, need to be versioned. BigFlow provides automatic versioning based on 
+the git tags system. There are 2 commands you need to know.
+
+The `bigflow project-version` command prints the current version of your project:
+
+```
+bigflow project-version
+>>> 0.34.0
+```
+
+As you can see, the version scheme is a standard Python package version scheme:
+
+`<major>.<minor>.<patch>`
+
+If BigFlow finds a tag on the current commit, it uses it as a current project version. If there are commits after the last tag,
+it creates a snapshot version with the following scheme:
+
+`<major>.<minor>.<patch><snapshot_id>`
+
+For example:
+
+```
+bigflow project-version
+>>> 0.34.0SHAdee9af83SNAPSHOT8650450a
+```
+
+If you are ready to release new version, you don't have to set a new tag manually. You can use the `bigflow release` command:
+
+```
+bigflow project-version
+>>> 0.34.0SHAdee9af83SNAPSHOT8650450a
+bigflow release
+bigflow project-version
+>>> 0.35.0
+```
+
+## Image
+
+To run a job in the desired environment, BigFlow makes use of Docker. Each job is executed from a Docker container, made
+from your project Docker image. The default image generated from the scaffolding tool looks like this:
+
+```dockerfile
+FROM python:3.7
+COPY ./dist /dist
+COPY ./resources /resources
+RUN apt-get -y update && apt-get install -y libzbar-dev libc-dev musl-dev
+RUN for i in /dist/*.whl; do pip install $i; done
+```
+
+As you can see, the basic image installs the generated Python package. With the installed package, you can run a workflow or a job
+from the Docker environment.
+
+Run the `bigflow build-image` command inside the `docs` project. Next, run the following command to run the example workflow from the docker:
+
+```shell script
+docker run bigflow-docs:0.1.0 bigflow run --job hello_world_workflow.hello_world
+```
+
+And that's how jobs are being run after deployment, from a generated DAG. 
```

Comment:
And that's how jobs are being run when deployed on Composer.
DAGs generated by Bigflow use `KubernetesPodOperator` to call this `docker` command.

 

`KubernetesPodOperator`

Reply:
done